### PR TITLE
Add changelog script to the workflow

### LINKED ISSUE #446: Break in smaller pieces current pipeline
As the pipeline has grown faster we might discuss here if we should break into smaller scripts or not.

fyi @dron22 @alanhamlett

### PR DESCRIPTION
**Pull request by gandarez**
This PR simplifies the current workflow by removing extra steps. In order to have a better changelog parsing a new script has been added.

Closes https://github.com/wakatime/wakatime-cli/issues/446

### CODE REVIEW

File: .github/workflows/on_push.yml [hunk 1]
```diff
@@ -218,7 +218,7 @@ jobs:
             github.git.deleteRef({
               owner: context.repo.owner,
               repo: context.repo.repo,
-              ref: "refs/tags/${{ needs.version.outputs.semver_tag }}"
+              ref: "tags/${{ needs.version.outputs.semver_tag }}"
```
Comment (alanhamlett) [inline .github/workflows/on_push.yml:246]: Why is this changing, will it work?

Comment (gandarez) [reply .github/workflows/on_push.yml:246]: It was incorrect actually. Got to know during the development of this task.

File: bin/prepare_changelog.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+slack_output_for_develop() {
+    local IFS=$'\n' # make newlines the only separator
+    local temp=
+    for j in $(echo "${changelog}")
+    do
+        hash=${j:0:7}
+        link="<https://github.com/wakatime/wakatime-cli/commit/${hash}|${hash}>"
+        temp="${temp}$(echo "$j" | awk '{printf "<https://github.com/wakatime/wakatime-cli/commit/"$1"|"$1">";$1=""; print $0 }')\n"
```
Comment (dron22) [inline bin/prepare_changelog.sh:32]: haha, nice to see the awk command again :sunglasses:

**Review by dron22 — APPROVED** [review]
LGTM :+1: As we had problems in the past, maybe it would make sense to cover changelog related logic with some tests? `prepare_changelog.sh` could tested here for example.

**Review by alanhamlett — APPROVED** [review]

**Comment by gandarez** [comment]
@dron22 I added testing for shell scripts and attached it to the CI as well.

**Pull request merged**